pkg/connector/source.go

```diff
@@ -201,16 +207,20 @@ func (s *Source) Ack(ctx context.Context, p record.Position) error {
 		return err
 	}
 
-	// lock to prevent race condition with connector persister
+	// lock as we are updating the state and leave it locked so the persister
+	// can safely prepare the connector before it stores it
```

Comment:
>  before it stores it

This is in reference to the persisted batch being flushed, and not that the connector will necessarily be stored here in Ack()?

Reply:
Yeah, by "prepare the connector" I mean that the connector will be encoded into bytes, and since it's locked already it should be safe to do so. The persister will then only store the encoded bytes in the batch and flush it at a later point in time.